### ISSUE TITLE
Cypress runner v8.0.0 crashes when clicking + New Spec File

### ISSUE THREAD

**Issue by wlsf82** [cypress-io/cypress#17430]
### Current behavior

I've initialized NPM with `npm init -y`, I've installed Cypress v8 as a dev dependency with `npm i cypress@8.0.0 -D`, and I've opened the Cypress runner with `npx cypress open`. Right after the runner has finished loading, I just clicked the <kbd>+ New Spec File</kbd>, then the runner crashed (closed automatically), and the following error was left in the terminal:

```
npm i cypress@8.0.0 -D

changed 1 package, and audited 171 packages in 1s

24 packages are looking for funding
  run `npm fund` for details

found 0 vulnerabilities
walmyr@Walmyrs-MacBook-Pro cyv8 % npx cypress open
TypeError: open_project_1.default.getConfig(...).then(...).tap is not a function
    at Object.showDialogAndCreateSpec (/Users/walmyr/Library/Caches/Cypress/8.0.0/Cypress.app/Contents/Resources/app/packages/server/lib/gui/files.js:18:10)
    at IpcMainImpl.handleEvent (/Users/walmyr/Library/Caches/Cypress/8.0.0/Cypress.app/Contents/Resources/app/packages/server/lib/gui/events.js:115:20)
    at apply (/Users/walmyr/Library/Caches/Cypress/8.0.0/Cypress.app/Contents/Resources/app/packages/server/node_modules/lodash/lodash.js:492:17)
    at IpcMainImpl.wrapper (/Users/walmyr/Library/Caches/Cypress/8.0.0/Cypress.app/Contents/Resources/app/packages/server/node_modules/lodash/lodash.js:5379:16)
    at IpcMainImpl.emit (events.js:315:20)
    at Object.<anonymous> (electron/js2c/browser_init.js:161:10092)
    at Object.emit (events.js:315:20)

TypeError: open_project_1.default.getConfig(...).then(...).tap is not a function
    at Object.showDialogAndCreateSpec (/Users/walmyr/Library/Caches/Cypress/8.0.0/Cypress.app/Contents/Resources/app/packages/server/lib/gui/files.js:18:10)
    at IpcMainImpl.handleEvent (/Users/walmyr/Library/Caches/Cypress/8.0.0/Cypress.app/Contents/Resources/app/packages/server/lib/gui/events.js:115:20)
    at apply (/Users/walmyr/Library/Caches/Cypress/8.0.0/Cypress.app/Contents/Resources/app/packages/server/node_modules/lodash/lodash.js:492:17)
    at IpcMainImpl.wrapper (/Users/walmyr/Library/Caches/Cypress/8.0.0/Cypress.app/Contents/Resources/app/packages/server/node_modules/lodash/lodash.js:5379:16)
    at IpcMainImpl.emit (events.js:315:20)
    at Object.<anonymous> (electron/js2c/browser_init.js:161:10092)
    at Object.emit (events.js:315:20)
```

> I tested the same steps with Cypress 7.7.0 (previous version of 8.0.0), and it worked fine.

### Desired behavior

The new test file should be created without crashing the runner.

### Test code to reproduce

Here is the repo with Cypress v8.0.0 and its `package-lock.json` file: https://github.com/wlsf82/cyv8

### Cypress Version

8.0.0

**Comment by jennifer-shehane** [comment]
What version were you updating from? Likely this was introduced in [7.6.0](https://on.cypress.io/changelog#7-6-0) where we introduced the deleting of examples files + create new spec button. 

Can you give specifics on your OS? What version of Mac are you using?

**Comment by panzarino** [comment]
Looks like we missed out on updating this one in the changes here - https://github.com/cypress-io/cypress/pull/17319. I'll submit a fix soon.

**Comment by panzarino** [comment]
Ok so that commit actually didn't make it into the release - this issue has actually been fixed in #17319 but that came a day after the release. I think this was the only instance where we chained a `.tap` off of the promise so it should be the only instance that's affected by this change.

Fix should be coming in the next release (either `8.0.1` or `8.1.0`)
